src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs

```diff
@@ -144,34 +144,36 @@ public async Task SpecializeAsync()
 
         private bool UsePlaceholderChannel(string workerRuntime)
         {
-            if (!string.IsNullOrEmpty(workerRuntime))
+            if (string.IsNullOrEmpty(workerRuntime))
             {
-                // Restart worker process if custom languageWorkers:[runtime]:arguments are passed in
-                var workerArguments = _config.GetSection($"{RpcWorkerConstants.LanguageWorkersSectionName}:{workerRuntime}:{WorkerConstants.WorkerDescriptionArguments}").Value;
-                if (!string.IsNullOrEmpty(workerArguments))
-                {
-                    return false;
-                }
+                return false;
+            }
 
-                // If a profile evaluates to true and was not previously loaded, restart worker process
-                if (!_profileManager.IsCorrectProfileLoaded(workerRuntime))
-                {
-                    return false;
-                }
+            // Restart worker process if custom languageWorkers:[runtime]:arguments are passed in
+            var workerArguments = _config.GetSection($"{RpcWorkerConstants.LanguageWorkersSectionName}:{workerRuntime}:{WorkerConstants.WorkerDescriptionArguments}").Value;
+            if (!string.IsNullOrEmpty(workerArguments))
+            {
+                return false;
+            }
 
-                // Special case: node and PowerShell apps must be read-only to use the placeholder mode channel
-                // Also cannot use placeholder worker that is targeting ~3 but has backwards compatibility with V2 enabled
-                // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
-                if (string.Equals(workerRuntime, RpcWorkerConstants.NodeLanguageWorkerName, StringComparison.OrdinalIgnoreCase)
-                    || string.Equals(workerRuntime, RpcWorkerConstants.PowerShellLanguageWorkerName, StringComparison.OrdinalIgnoreCase)
-                    || string.Equals(workerRuntime, RpcWorkerConstants.PythonLanguageWorkerName, StringComparison.OrdinalIgnoreCase))
-                {
-                    // Use if readonly and not v2 compatible on ~3 extension
-                    return _applicationHostOptions.CurrentValue.IsFileSystemReadOnly && !_environment.IsV2CompatibileOnV3Extension();
-                }
-                return true;
+            // Special case: node and PowerShell apps must be read-only to use the placeholder mode channel
+            // Also cannot use placeholder worker that is targeting ~3 but has backwards compatibility with V2 enabled
+            // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
+            if (string.Equals(workerRuntime, RpcWorkerConstants.NodeLanguageWorkerName, StringComparison.OrdinalIgnoreCase)
```

Comment:
Not sure if you are interested in pattern matching, but you could accomplish this that way. Only drawback is it has to be constants and only ordinal works. Then you get into the issue of normalizing to upper case over lower case (Turkish i issue, but that _really_ isn't applicable here - so lowercase would be safe).

``` CSharp
if (workerRuntime.ToUpperInvariant() is "NODE" or "POWERSHELL" or "PYTHON")
```

Reply:
Doing it this way should still be okay I assume?

```
if (workerRuntime.ToLowerInvariant() is RpcWorkerConstants.NodeLanguageWorkerName
                or RpcWorkerConstants.PowerShellLanguageWorkerName
                or RpcWorkerConstants.PythonLanguageWorkerName)
```

Reply:
Won't the call to `ToUpperInvariant` or `ToLowerInvariant` here for the pattern matching approach cause a string copy/allocation and overall perform worse?

Reply:
I believe Fabio did a benchmark on this and the existing approach is more performant - will revert this change in the next PR